src/main/controllers/price-service.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const EventEmitter = require('events');
+const fetch = require('node-fetch');
+
+const TokenPrice = require('./models/token-price');
+
+const emitter = new EventEmitter();
+
+const loadPriceData = async () => {
+	const response = await fetch('https://coincap.io/front');
```

Comment:
@dfcowell I was thinking to use the streaming api https://github.com/CoinCapDev/CoinCap.io#socket-api. so we won't need a refresh and we could even not need to store it on the DB as the wallet cannot run offline. Thoughts? cheers

Reply:
I like the idea of the streaming API however we should keep a local cached copy in order to speed up wallet start time (we don't want to have to wait to download the token list or receive it through the stream before we can render balance info) and provide gracefully degraded functionality in the event that the CoinCap API goes down.

Reply:
Agreed about caching locally though. Do you know if coins update all together? If yes, there is no saving burden when editing on the DB. Otherwise if they change separated it should increase performance as we don't need to edit 13 hundreds records at once.